homeassistant/components/nut/device_action.py

```diff
@@ -0,0 +1,113 @@
+"""Provides device actions for Network UPS Tools (NUT)."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_TYPE
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.helpers import device_registry
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType, TemplateVarsType
+
+from . import PyNUTData
+from .const import DEVICE_SUPPORTED_COMMANDS, DOMAIN, PYNUT_DATA
+
+ACTION_TYPES = {
```

Comment:
Can we generate this from INTEGRATION_SUPPORTED_COMMANDS by mutating that data?

Reply:
That was my original intent, since it feels duplicated. After a few tries I gave up because I thought that `ACTION_TYPES` was expected to be a constant and that I could not generate the constant using a mutation. I am fairly new to Python and totally new to Home Assistant development, so I could be totally wrong here.

As a side note, with `INTEGRATION_SUPPORTED_COMMANDS` I was also aiming at filtering out those that receive parameters, since the library used does not support params. I could try to submit a PR later on if we feel it makes sense to support those two missing commands, but I am unclear of whether the library is still maintained.

By the way, I am still working on a few tests and may revisit some of the code, but your feedback is greatly appreciated.

Reply:
Untested but I think you could write it as 
```diff
diff --git a/homeassistant/components/nut/device_action.py b/homeassistant/components/nut/device_action.py
index 1cd6d5275c..a288fbd781 100644
--- a/homeassistant/components/nut/device_action.py
+++ b/homeassistant/components/nut/device_action.py
@@ -10,40 +10,15 @@ import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.typing import ConfigType, TemplateVarsType
 
 from . import PyNUTData
-from .const import DEVICE_SUPPORTED_COMMANDS, DOMAIN, PYNUT_DATA
-
-ACTION_TYPES = {
-    "beeper_disable",
-    "beeper_enable",
-    "beeper_mute",
-    "beeper_toggle",
-    "bypass_start",
-    "bypass_stop",
-    "calibrate_start",
-    "calibrate_stop",
-    "load_off",
-    "load_on",
-    "outlet_n_load_cycle",
-    "outlet_n_load_off",
-    "outlet_n_load_on",
-    "outlet_n_shutdown_return",
-    "reset_input_minmax",
-    "reset_watchdog",
-    "shutdown_reboot",
-    "shutdown_reboot_graceful",
-    "shutdown_return",
-    "shutdown_stayoff",
-    "shutdown_stop",
-    "test_battery_start",
-    "test_battery_start_deep",
-    "test_battery_start_quick",
-    "test_battery_stop",
-    "test_failure_start",
-    "test_failure_stop",
-    "test_panel_start",
-    "test_panel_stop",
-    "test_system_start",
-}
+from .const import (
+    DEVICE_SUPPORTED_COMMANDS,
+    DOMAIN,
+    PYNUT_DATA,
+    INTEGRATION_SUPPORTED_COMMANDS,
+)
+
+ACTION_TYPES = {cmd.replace(".", "_") for cmd in INTEGRATION_SUPPORTED_COMMANDS}
+
 
 ACTION_SCHEMA = cv.DEVICE_ACTION_BASE_SCHEMA.extend(
     {

```

Reply:
Apologies for the radio silence, @bdraco. Your proposal works like a charm. The PR should be good for another pass. Thanks!